shap/plots/_bar.py

```diff
@@ -341,7 +341,7 @@ def bar(shap_values, max_display=10, order=Explanation.abs, clustering=None, clu
 
     if show:
         pl.show()
-
+    return pl.gcf()
```

Comment:
_waveform.py returns `pl.gcf()` when `show == False`. Does it make sense to always return `pl.gcf()?`

Reply:
@dsgibbons I think I see the intention, which is that when `show=False`, it means the user is likely to take the plot and customize it further. So returning *something* makes sense to me.

I'm wondering whether we should return the Figure or the Axes. (If this would turn into a larger discussion, I suggest leaving this change out of this PR since it would hinder the PR being merged).

I'm thinking Axes, as I'm hoping our shap plots to be based around Ax rather than Figs in the future. (we can discuss reasons for this in #3070).

Reply:
I prefer Ax too. I chose `pl.gcf()` because it is already used in the waterfall diagram. Migrating to Ax sounds like a broader API change - perhaps for 0.44?

Reply:
@dsgibbons That's true, let's stick to the current api for now then (for consistency). I'll work on the PR to migrate to axes for 0.44.